pkg/trace/stats/sublayers.go

```diff
@@ -34,6 +43,181 @@ func (v SublayerValue) GoString() string {
 	return v.String()
 }
 
+// SublayerCalculator holds arrays used to compute sublayer metrics.
```

Comment:
Perhaps this is a great opportunity to answer the (what for many is a) mystery question:
* What are sublayer metrics?

I've been asked a lot of time by other developers, who have also looked at trace-agent code. This would help in that case.

It would be nice to add more here, and explain how each span is represented in each array as an index etc... Explain the algorithm.